stock_quant_merge/models/stock.py

```diff
@@ -19,14 +19,20 @@ def _mergeable_domain(self):
                 ('lot_id', '=', self.lot_id.id),
                 ('package_id', '=', self.package_id.id),
                 ('location_id', '=', self.location_id.id),
-                ('reservation_id', '=', False),
+                ('reservation_id', '=', self.reservation_id.id),
```

Comment:
I think if we merge quants with different "in date" values, we'll be introducing wrong results in the FIFO/LIFO removal strategy.